cpp/util/libevent_wrapper.cc

```diff
@@ -76,12 +80,71 @@ struct HttpServer::Handler {
 };
 
 
-Base::Base()
+class ResolverImpl : public Base::Resolver {
+ public:
+  string Resolve(const string& host) override {
+    struct addrinfo* info;
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    // It seems evhtp doesn't support IPv6 addresses because it uses
+    // inet_addr() to parse the passed in "stringified" address.
+    // Restrict to IPv4 addresses for now:
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    const int resolved(getaddrinfo(host.c_str(), AF_UNSPEC, &hints, &info));
+    if (resolved != 0) {
+      LOG(WARNING) << "Failed to resolve HTTPS hostname " << host << ": "
+                   << gai_strerror(resolved);
+      return nullptr;
+    }
+
+    struct addrinfo* res(info);
+    void* addr(nullptr);
+    while (res && !addr) {
+      switch (res->ai_family) {
+        case AF_INET:
+          addr =
+              &reinterpret_cast<struct sockaddr_in*>(res->ai_addr)->sin_addr;
+          break;
+        case AF_INET6:
+          // Just in case one day evhtp uses inet_pton()
+          addr =
+              &reinterpret_cast<struct sockaddr_in6*>(res->ai_addr)->sin6_addr;
+          break;
+        default:
+          res = res->ai_next;
+          break;
+      }
+    }
+
+    if (!addr) {
+      LOG(WARNING) << "Got no usable address for " << host;
+      return nullptr;
+    }
+
+    char addr_str[INET6_ADDRSTRLEN];
+    inet_ntop(res->ai_family, addr, addr_str, INET6_ADDRSTRLEN);
+    freeaddrinfo(info);
+    return string(addr_str);
+  }
+};
+
+
+Base::Base() : Base(unique_ptr<Resolver>(new ResolverImpl)) {
```

Comment:
Oh, because you're passing it to the other constructor here, I thought you were initializing `resolver_` directly (but this is much better!).